content/en/docs/ambient/extensibility/wasm-module-distribution/index.md

```diff
@@ -0,0 +1,162 @@
+
+To configure a WebAssembly filter with a remote Wasm module, create a `WasmPlugin` resource targeting the `bookinfo-gateway`:
+
+{{< text bash >}}
+$ kubectl get gateway
+NAME               CLASS            ADDRESS        PROGRAMMED   AGE
+bookinfo-gateway   istio            172.18.7.110   True         23h
+{{< /text >}}
+
+{{< text bash >}}
+$ kubectl apply -f - <<EOF
+apiVersion: extensions.istio.io/v1alpha1
+kind: WasmPlugin
+metadata:
+  name: basic-auth-at-gateway
+spec:
+  targetRef:
+    group: gateway.networking.k8s.io
+    kind: Gateway
+    name: bookinfo-gateway # gateway name retrieved from previous step
+  url: oci://ghcr.io/istio-ecosystem/wasm-extensions/basic_auth:1.12.0
+  phase: AUTHN
+  pluginConfig:
+    basic_auth_rules:
+      - prefix: "/productpage"
+        request_methods:
+          - "GET"
+          - "POST"
+        credentials:
+          - "ok:test"
+          - "YWRtaW4zOmFkbWluMw=="
+EOF
+{{< /text >}}
+
+An HTTP filter will be injected at the gateway as an authentication filter.
+The Istio agent will interpret the WasmPlugin configuration, download remote Wasm modules from the OCI image registry to a local file, and inject the HTTP filter at the gateway by referencing that file.
+
+### Verify the traffic via the Gateway
+
+1. Test `/productpage` without credentials
+
+    {{< text bash >}}
+    $ kubectl exec deploy/sleep -- curl -s -w "%{http_code}" -o /dev/null "http://$GATEWAY_HOST/productpage"
+    401
+    {{< /text >}}
+
+1. Test `/productpage` with credentials
+
+    {{< text bash >}}
+    $ kubectl exec deploy/sleep -- curl -s -o /dev/null -H "Authorization: Basic YWRtaW4zOmFkbWluMw==" -w "%{http_code}" "http://$GATEWAY_HOST/productpage"
+    200
+    {{< /text >}}
+
+## Apply Wasm Configuration at Waypoint Proxy
+
+### Deploy a waypoint proxy
+
+Follow the [waypoint deployment instructions](docs/ambient/getting-started/#layer-7-authorization-policy) to deploy a waypoint proxy in the bookinfo namespace.
+
+{{< text bash >}}
+$ istioctl x waypoint apply --enroll-namespace --wait
+{{< /text >}}
+
+### Verify traffic without WasmPlugin at the waypoint
+
+{{< text bash >}}
+$ kubectl exec deploy/sleep -- curl -s -w "%{http_code}" -o /dev/null http://productpage:9080/productpage
+200
+{{< /text >}}
+
+### Apply WasmPlugin at waypoint proxy
+
+To configure a WebAssembly filter with a remote Wasm module, create a `WasmPlugin` resource targeting the `waypoint` gateway:
+
+{{< text bash >}}
+$ kubectl get gateway
+NAME               CLASS            ADDRESS        PROGRAMMED   AGE
+bookinfo-gateway   istio            172.18.7.110   True         23h
+waypoint           istio-waypoint   10.96.202.82   True         21h
+{{< /text >}}
+
+{{< text bash >}}
+$ kubectl apply -f - <<EOF
+apiVersion: extensions.istio.io/v1alpha1
+kind: WasmPlugin
+metadata:
+  name: basic-auth-at-waypoint
+spec:
+  targetRef:
+    group: gateway.networking.k8s.io
+    kind: Gateway
```

Comment:
should do another task on target a given service imho.  No need to give tasks on ingress gateway as it is same as sidecars.